streams/src/main/java/org/apache/kafka/streams/KafkaClientSupplier.java

```diff
@@ -31,29 +30,14 @@
  * @see KafkaStreams#KafkaStreams(Topology, java.util.Properties, KafkaClientSupplier)
  */
 public interface KafkaClientSupplier {
-    /**
-     * Create an {@link AdminClient} which is used for internal topic management.
-     *
-     * @param config Supplied by the {@link java.util.Properties} given to the {@link KafkaStreams}
-     * @return an instance of {@link AdminClient}
-     * @deprecated Not called by Kafka Streams, which now uses {@link #getAdmin} instead.
-     */
-    @Deprecated
-    default AdminClient getAdminClient(final Map<String, Object> config) {
-        throw new UnsupportedOperationException("Direct use of this method is deprecated. " +
-            "Implementations of KafkaClientSupplier should implement the getAdmin() method instead. " +
-            "The method will be removed in a future release.");
-    }
-
     /**
      * Create an {@link Admin} which is used for internal topic management.
      *
      * @param config Supplied by the {@link java.util.Properties} given to the {@link KafkaStreams}
      * @return an instance of {@link Admin}
      */
-    @SuppressWarnings("deprecation")
     default Admin getAdmin(final Map<String, Object> config) {
-        return getAdminClient(config);
+        throw new UnsupportedOperationException("Implementations of KafkaClientSupplier should implement the getAdmin() method.");
```

Comment:
Would we want to remove this default implementation too? Not saying we should, but what's the benefit?

Reply:
I think we can; I was originally worried if I change it from `default` to a pure interface would that be binary compatible --- did some research and it seems okay.